module/VuFindApi/src/VuFindApi/Controller/SearchApiController.php

```diff
@@ -304,89 +336,161 @@ public function searchAction()
         $request = $this->getRequest()->getQuery()->toArray()
             + $this->getRequest()->getPost()->toArray();
 
+        $isCursorSearch = ($request['resumptionToken'] ?? false);
+        try {
+            if ($isCursorSearch) {
+                $response = $this->doCursorSearch($request);
+            } else {
+                $response = $this->doDefaultSearch($request);
+            }
+        } catch (Exception $e) {
+            return $this->output([], self::STATUS_ERROR, 400, $e->getMessage());
+        }
+        return $this->output($response, self::STATUS_OK);
+    }
+
+    /**
+     * Perform a search using page in Solr
+     *
+     * @param array $request Array containing combination of post and get request params
+     *
+     * @return array Response to be sent for the user
+     *               - records: Records found
+     *               - resultCount: Total result count
+     *               - facets: array containing facets for the result
+     */
+    protected function doDefaultSearch(array $request): array
+    {
         if (
-            isset($request['limit'])
-            && (!ctype_digit($request['limit'])
+            isset($request['limit']) && (!ctype_digit($request['limit'])
             || $request['limit'] < 0 || $request['limit'] > $this->maxLimit)
         ) {
-            return $this->output([], self::STATUS_ERROR, 400, 'Invalid limit');
-        }
-
-        // Sort by relevance by default
-        if (!isset($request['sort'])) {
-            $request['sort'] = 'relevance';
+            throw new Exception('Invalid limit', 400);
         }
-
-        $requestedFields = $this->getFieldList($request);
-
-        $facetConfig = $this->getConfig('facets');
-        $hierarchicalFacets = isset($facetConfig->SpecialFacets->hierarchical)
-            ? $facetConfig->SpecialFacets->hierarchical->toArray()
-            : [];
-
-        $runner = $this->getService(\VuFind\Search\SearchRunner::class);
-        try {
-            $results = $runner->run(
-                $request,
-                $this->searchClassId,
-                function (
-                    $runner,
-                    $params,
-                    $searchId
-                ) use (
-                    $hierarchicalFacets,
-                    $request,
-                    $requestedFields
-                ) {
-                    foreach ($request['facet'] ?? [] as $facet) {
-                        if (!isset($hierarchicalFacets[$facet])) {
-                            $params->addFacet($facet);
-                        }
-                    }
-                    if ($requestedFields) {
-                        $limit = $request['limit'] ?? 20;
-                        $params->setLimit($limit);
-                    } else {
-                        $params->setLimit(0);
+        $limit = $request['limit'] ??= 20;
+        $request['sort'] ??= self::DEFAULT_SORT;
```

Comment:
Is there a reason not to use the default sort from the search options object, and/or to make this configurable somehow? Using a constant seems like the least flexible option.

Reply:
Hmm, if I understood properly and just by leaving the removal out, it should use the default sort itself without having to assign it?